docs/packages/context-menu/Context-menu-manager.md

```diff
@@ -0,0 +1,345 @@
+The context menu manager provides a convenient way for managing the context menu structure.
+The manager is responsible for building a tree of nested groups/items that forms the actual structure of the context menu, as presented in the UI.   
+The manager is based on CRUD principles while providing simple functions for adding, updating, deleting, and in the end - provides a function for reading the main tree that have been created.
+
+### Context menu UI behavior
+For better understanding the structure and the manager, lets see some important UI notes:
+
+* The context menu displays a list of items, by default each item provides an action when clicking on it.
+
+* Items can provide action also on hover (mouseEnter).
+
+* If an item contains children groups, hovering on it will open a sub menu, and clicking on it won't trigger action.
+
+* Groups themselves are not visually displayed, they just displays their items and separate between them.
+
+* A group can display a title, if a title is specified.
+
+* Items of a group are displayed together, separated from other group items with a line separator.
+
+* Nested groups are displayed as a sub-menu aside the main context-menu.
+
+### Context Menu config structure
+The context menu config structure is built as a tree of groups & items.
+The root is based on groups, each group contains one or more item (inside its `items[]` property).
+Items can also have children, inside their `groups[]` property. those children are *nested groups*.
+
+### Structure tree
+
+```TS
+interface Group {
+	id: string; 				// Must be unique.
+	title: string;				// If exists, displayed as a group title without action to trigger.
+	disabled: Function | boolean;		// If true, the entire group is Displayed but disabled.
+	parentItemId: string;			// Id of the parent (a valid item id). if used - creates a nested menu.
+	items: Item[]; 				// List of items.
+};
+
+interface Item {
+	id: string;				// Must be unique.
+	label: string;				// Describes the action.
+	icon: Function | string;		// Can be a callback for checkbox effect.
+	shortCut: string;			// ShortCut symbols. 
+	badge: Badge;				// Creates a visual badge for special indications.
+	disabled: Function | boolean;		// If true, displayed but disabled.
+	action: Function;			// A callback to be triggered on-click.
+	onMouseEnter: Function;			// A callback to be triggered on-mouse-enter.
+	onMouseLeave: Function;			// A callback to be triggered on-mouse-leave.
+	groups: Group[];			// List of groups (for nested menus).
+};
+
+interface Badge {
+	backgroundColor: string;		// Background color of the badge.
+	color: string;				// Text color of the badge.
+	text: string;				// Text of badge.
+}
+
+type Structure = Group[];
+```
+
+
+&nbsp;  
+## API methods
+Each method has parameters explanation and attention notes if needed.
+
+&nbsp;
+### Create a group
+`createGroup( newGroup )` Method creates a group and add it to the config tree. 
+
+&nbsp;  
+Example:
+```JS
+createGroup(
```

Comment:
```suggestion
manager.createGroup(
```

Reply:
I'm not sure its important here. I don't explain how to use it (why 'manager'?, it can be also API or many things...) but just show the function parameters in use.

Reply:
because then it looks like some global function

Reply:
I think if we add 'manager' we need also to add a lino for  
const manager = new ContextMenuManager();
so I prefer not to. I leave it like this.